metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java

```diff
@@ -0,0 +1,180 @@
+/*
+ *       Copyright 2017 Netflix, Inc.
+ *          Licensed under the Apache License, Version 2.0 (the "License");
+ *          you may not use this file except in compliance with the License.
+ *          You may obtain a copy of the License at
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *          Unless required by applicable law or agreed to in writing, software
+ *          distributed under the License is distributed on an "AS IS" BASIS,
+ *          WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *          See the License for the specific language governing permissions and
+ *          limitations under the License.
+ */
+
+package com.netflix.metacat.connector.hive;
+
+import com.netflix.metacat.common.server.util.DataSourceManager;
+import com.netflix.metacat.common.server.util.MetacatConnectorProperties;
+import com.netflix.metacat.connector.hive.client.DefaultNoOpHiveClient;
+import com.netflix.metacat.connector.hive.converters.HiveConnectorInfoConverter;
+import com.netflix.metacat.connector.hive.converters.HiveTypeConverter;
+import com.netflix.metacat.common.server.util.JdbcUtil;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+/**
+ * Hive configs.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+@Configuration
```

Comment:
We should talk about this. I think it's going to be a problem due to classpath scanning. Boot by default launches with classpath scanning for everything under `com.netflix.metacat` so this class will be picked up by the main spring app context when what we really want is it to only be picked up when a connector is instantiated on demand. You shouldn't have to mark all the beans as `prototype` as you'd want them to be singletons within that connector instance spring context. In this current manner every time a bean is requested it will create a new one which is not desirable. I'm not currently sure the best way to do this but we can research it.